torch/csrc/jit/passes/lower_tuples.cpp

```diff
@@ -107,85 +176,106 @@ static void RemoveTupleConstants(Node* n) {
   n->replaceAllUsesWith(tuple_construct);
 }
 
-static void VisitNode(Node* n, Node* insert_point) {
-  auto& graph = *n->owningGraph();
-
-  // tuple construction operators will become dead when the unpacks are replaced
-  if (n->kind() == prim::TupleConstruct) {
-    return;
-  }
-
-  // note: changing the second argument to false changes this pass from a
-  // complete lowering pass to one that removes tuples when possible. When
-  // tuples are first-class in the interpreter, we should still run this pass to
-  // remove extraneous uses
-
-  if (n->kind() == prim::TupleUnpack || n->kind() == prim::TupleIndex ||
-      n->kind() == prim::TupleSlice) {
-    removeTupleNodes(n, /*must_remove_tuples*/ true);
-    return;
-  }
-
+static void flattenInputs(Node* n, Node* insert_point) {
   // flatten the input list  op(a, tup, b) --> op(a, t0, t1, b)
   for (size_t i = 0; i < n->inputs().size();) {
     auto input = n->inputs()[i];
     if (TupleTypePtr tt = input->type()->cast<TupleType>()) {
       TORCH_CHECK(
-          supported_ops.count(n->kind()) > 0,
-          "tuple appears in op that does not forward tuples, ",
-          "unsupported kind: ",
+          (input->node()->kind() == prim::TupleConstruct),
+          "tuple use not matched to tuple construct. Instead found: ",
           n->kind().toQualString());
-      TORCH_CHECK(
-          input->node()->kind() == prim::TupleConstruct,
-          "tuple use not matched to tuple construct");
-      for (size_t j = 0; j < tt->elements().size(); ++j) {
-        n->insertInput(i + 1 + j, input->node()->inputs().at(j));
+      if (supported_ops.count(n->kind()) > 0) {
+        if ((n->kind() == prim::Loop)) {
+          if (input->node()->kind() == prim::TupleConstruct) {
+            flattenTupleInBlockParam(n, i);
```

Comment:
Is this function only suitable for `prim::Loop`? Then we may rename the function name to indicate that...

Reply:
This can be used for any blocks with tuple inputs. Renamed it for now.